hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemDelete.java

```diff
@@ -130,4 +146,49 @@ public Void call() throws Exception {
     assertPathDoesNotExist(fs, "deleted", dir);
 
   }
+
+  @Test
+  public void testDeleteIdempotency() throws Exception {
+    org.junit.Assume.assumeTrue(DEFAULT_DELETE_CONSIDERED_IDEMPOTENT);
+    // Config to reduce the retry and maxBackoff time for test run
+    AbfsConfiguration abfsConfig = getConfiguration();
+    abfsConfig.setMaxIoRetries(reducedRetryCount);
+    abfsConfig.setMaxBackoffIntervalMilliseconds(reducedMaxBackoffIntervalMs);
+
+    final AzureBlobFileSystem fs = getFileSystem();
+    AbfsClient abfsClient = fs.getAbfsStore().getClient();
+    AbfsPerfTracker tracker = new AbfsPerfTracker("test",
+        this.getAccountName(),
+        abfsConfig);
+
+    // Create test AbfsClient
+    AbfsClient testClient = new AbfsClient(
+        abfsClient.getBaseUrl(),
+        new SharedKeyCredentials(abfsConfig.getAccountName().substring(0,
+            abfsConfig.getAccountName().indexOf(DOT)),
+            abfsConfig.getStorageAccountKey()),
+        abfsConfig,
+        new ExponentialRetryPolicy(reducedRetryCount),
+        abfsConfig.getTokenProvider(),
+        tracker);
+
+    // Mock instance of AbfsRestOperation
+    AbfsRestOperation op = mock(AbfsRestOperation.class);
+    // Set retryCount to non-zero
+    when(op.getRetryCount()).thenReturn(reducedRetryCount);
+
+    // Mock instance of Http Operation response. This will return HTTP:Not Found
+    AbfsHttpOperation http404Op = mock(AbfsHttpOperation.class);
+    when(http404Op.getStatusCode()).thenReturn(HTTP_NOT_FOUND);
+
+    // Mock delete response to 404
+    when(op.getResult()).thenReturn(http404Op);
+
+    assertTrue(
```

Comment:
Try to use the better alternative assertJ.

Reply:
Fixed